docs/logger.rst

```diff
@@ -0,0 +1,227 @@
+======
+Logger
+======
+
+``#include "util/Logger.h"``
+
+The BDTK Logger is based on `Boost.Log`_ with a design goal of being largely, though not completely, backward
+compatible with `glog`_ in usage, but with additional control over the logging format and other features.
+
+.. _Boost.Log: https://www.boost.org/libs/log/doc/html/index.html
+.. _glog: https://github.com/google/glog
+
+For the developer, log entries are made in a syntax that is similar to ``std::ostream``. Example::
+
+    LOG(INFO) << "x = " << x;
+
+where ``INFO`` is one common example of the log "severity" level. Other severities are ``WARNING``, ``FATAL``,
+``DEBUG1``, etc. See `Severity`_.
+
+In addition, there are a number of ``CHECK`` macros which act like ``assert()`` but will report via the
+logging system upon failure, with an optional message, and subsequently call ``abort()``. Examples::
+
+    CHECK(ptr);
+    CHECK_LT(1u, list.size()) << "list must contain more than 1 element.";
+
+If ``ptr==nullptr`` or ``1u >= list.size()`` then the program will abort, and log a corresponding ``FATAL`` message.
+See `CHECK`_.
+
+Initialization and Global Instances
+-----------------------------------
+
+To initialize the logging system, the function::
+
+    namespace logger {
+
+    void init(LogOptions const&);
+
+    }
+
+must be invoked with the ``logger::LogOptions`` object to be applied. It is recommended to run this from
```

Comment:
how to use LogOptions？

Reply:
logger::LogOptions log_options(argv[0]);

Reply:
I mean which log options we can set